pulsar/negative_acks_tracker.go

```diff
@@ -76,14 +95,48 @@ func (t *negativeAcksTracker) Add(msgID messageID) {
 	t.negativeAcks[batchMsgID] = targetTime
 }
 
-func (t *negativeAcksTracker) track() {
+func (t *negativeAcksTracker) AddMessage(msg Message) {
```

Comment:
Why is there a new method here?

Also, it looks like state is changing here without a lock. If multiple go routines call this at once multiple tracking routines could be started right? 

Can the tracking go routine just be started at creation time?

Reply:
Because we need to get redeliveryCount through the Message interface